pkg/cmd/generate-bazel-extra/main_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"bytes"
+	"encoding/gob"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseQueryXML(t *testing.T) {
+	testDataDir := testutils.TestDataPath(t, "TestParseQueryXML")
+	files, err := os.ReadDir(testDataDir)
+	require.NoError(t, err)
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".expected") {
+			continue
+		}
+		xmlData, err := os.ReadFile(filepath.Join(testDataDir, file.Name()))
+		require.NoError(t, err)
+		sizeToTargets, err := parseQueryXML(xmlData)
+		require.NoError(t, err)
+		expectedData, err := os.ReadFile(filepath.Join(testDataDir, strings.Split(file.Name(), ".xml")[0]+".expected"))
+		require.NoError(t, err)
+		var expectedDataMap map[string][]string
+		require.NoError(t, gob.NewDecoder(bytes.NewReader(expectedData)).Decode(&expectedDataMap))
```

Comment:
The test is a little odd because the `tc1.expected` file is an unreadable binary file. Ideally I would be able to read the test and its `testdata` files and come away with a pretty good understanding of what the test is actually testing. Since `tc1.expected` isn't readable, I come away from this test not knowing what we're actually testing for.

Can you just make this a literal `map[string][]string` in the source code? That will be far more readable.

Since there's only one test file (`tc1.xml`) it seems sensible to hard-code the path to `tc1.xml` and the expected value rather than iterating through every file in `testdata` unnecessarily when there's definitely only one relevant file.

Reply:
> Ideally I would be able to read the test and its testdata files and come away with a pretty good understanding of what the test is actually testing.

Makes sense, done.

Reply:
Way better, thanks.